Sources/FileSystem/Configuration/TestflightConfigurationProcessor.swift

```diff
@@ -1,22 +1,83 @@
 // Copyright 2020 Itty Bitty Apps Pty Ltd
 
+import CodableCSV
 import Foundation
 import Model
+import Files
+import Yams
 
-public struct TestflightConfigurationProcessor: ResourceWriter {
+public struct TestflightConfigurationProcessor {
 
-    let path: ResourcePath
+    let appsFolderPath: String
 
-    public init(path: ResourcePath) {
-        self.path = path
+    public init(appsFolderPath: String) {
+        self.appsFolderPath = appsFolderPath
     }
 
     public func writeConfiguration(
         apps: [Model.App],
         testers: [Model.BetaTester],
         groups: [Model.BetaGroup]
-    ) {
-        // TODO: Convert models to a TestflightConfiguration and write it to disk
+    ) throws {
+        // Generate configuration
+        let groupsByApp = Dictionary(grouping: groups, by: \.app?.id)
+
+        let configurations: [TestflightConfiguration] = apps.map { app in
+            var config = TestflightConfiguration(app: app)
+
+            config.betaTesters = testers
+                .filter { tester in tester.apps.map(\.id).contains(app.id) }
+                .compactMap(FileSystem.BetaTester.init)
+
+            config.betaGroups = (groupsByApp[app.id] ?? []).map { betaGroup in
+                FileSystem.BetaGroup(
+                    betaGroup: betaGroup,
+                    betaTesters: testers.filter { $0.betaGroups.map(\.id).contains(betaGroup.id) }
+                )
+            }
+
+            return config
+        }
+
+        // Write out the configuration
+        let appsFolder = try Folder(path: appsFolderPath)
+        try appsFolder.delete()
+
+        let rowsForTesters: ([BetaTester]) -> [[String]] = { testers in
+            let headers = [BetaTester.CodingKeys.allCases.map(\.rawValue)]
+            let rows = testers.map { [$0.email, $0.firstName, $0.lastName] }
+            return headers + rows
+        }
+
+        let filenameForBetaGroup: (BetaGroup) -> String = { betaGroup in
+            return betaGroup.groupName
+                .components(separatedBy: CharacterSet(charactersIn: " *?:/\\."))
+                .joined(separator: "_")
+                + ".yml"
```

Comment:
How about extracting the naming logic to be a var or helper in extension to `BetaGroup`?
```
extension FileSystem.BetaGroup {
    var filename: String { xxx }
}
```

Reply:
Since it's only used in one place I didn't see a reason at the moment to share it across the module